docs/content/guides/dagster-pipes/subprocess/modify-external-code.mdx

```diff
@@ -0,0 +1,248 @@
+
+#### Report asset materialization
+
+Similar to [reporting materialization metadata within the Dagster process](/concepts/assets/software-defined-assets#recording-materialization-metadata), you can also report asset materialization back to Dagster from the external process.
+
+In this example, we’re passing a piece of metadata named `total_orders` to the `metadata` parameter of the <PyObject module="dagster_pipes" object="PipesContext" method="report_asset_materialization" />. This payload will be sent from the external process back to Dagster:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/external_code_with_events.py startafter=start_events_marker endbefore=end_events_marker dedent=4
+# get the Dagster Pipes context
+context = PipesContext.get()
+# send structured metadata back to Dagster
+context.report_asset_materialization(metadata={"total_orders": total_orders})
+```
+
+Then, total_orders will show up in the UI as structured metadata:
+
+<Image
+  alt="Report asset materialization to Dagster"
+  src="/images/guides/dagster-pipes/modifying-code-5-1-report-asset-materialization.png"
+  width={1412}
+  height={806}
+/>
+
+This metadata will also be displayed on the Events tab of the Asset Details page in the UI:
+
+<Image
+  alt="View materialization events in asset details page"
+  src="/images/guides/dagster-pipes/modifying-code-5-2-materialization-asset-details.png"
+  width={1286}
+  height={736}
+/>
+
+#### Report asset checks
+
+Dagster allows you to define and execute data quality checks on assets. Refer to the [Asset Checks](/concepts/assets/asset-checks) documentation for more information.
+
+If your asset has data quality checks defined, you can report to Dagster that an asset check has been performed via <PyObject module="dagster_pipes" object="PipesContext" method="report_asset_check" />:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/external_code_with_events.py startafter=start_checks_marker endbefore=end_checks_marker dedent=4
+# report data quality check result back to Dagster
+context.report_asset_check(
+    passed=orders_df[["item_id"]].notnull().all().bool(),
+    check_name="no_empty_order_check",
+)
+```
+
+When Dagster executes the code, you’ll see an asset check event with the check result in the UI:
+
+<Image
+  alt="Report asset checks to Dagster"
+  src="/images/guides/dagster-pipes/modifying-code-5-3-report-asset-checks.png"
+  width={1312}
+  height={734}
+/>
+
+This check result will also be displayed on the Checks tab of the Asset Details page in the UI:
+
+<Image
+  alt="View checks in asset details page"
+  src="/images/guides/dagster-pipes/modifying-code-5-4-checks-asset-details.png"
+  width={1302}
+  height={730}
+/>
+
+---
+
+At this point, your two files should look like the following:
```

Comment:
```suggestion
## Finished code

At this point, your two files should look like the following:
```

I think this part could use a heading - not tied to this one specifically though